text_formatter.go

```diff
@@ -74,8 +77,19 @@ func (f *TextFormatter) Format(entry *Entry) ([]byte, error) {
 
 	prefixFieldClashes(entry.Data)
 
-	isColorTerminal := isTerminal && (runtime.GOOS != "windows")
-	isColored := (f.ForceColors || isColorTerminal) && !f.DisableColors
+	f.terminalLock.RLock()
```

Comment:
I think you can use a `sync.Once` here to make this a bit less complicated:

```
f.terminalOnce.Do(func() {
  f.isTerminal = IsTerminal(entry.Logger.Out)
})
```

Once `f.isTerminal` is determined and guarded by this statement, it should be safe to read. The cost will be reduced to the fast case for `sync.Once`.